### ISSUE TITLE
About ranking loss and EuclideanLoss

### ISSUE THREAD

**Issue by irvingzhang0512** [abhimanyudubey/dlcity#1]
Hi, thank you for your paper and code.

I read the paper and try to understand the model in `train_val.prototxt` .
As far as i know, `final1` & `final2` are both ranking functions,  and `labelr` produces labels of {-1, 1}, then the ranking_loss is `EuclideanLoss((final1 - final2) * labelr, 0)`

my question is:
1. For me, `EuclideanLoss((final1 - final2) * labelr, 0)` is quite strange, because `labelr` seems to be useless in this loss function.
2.  `EuclideanLoss((final1 - final2) * labelr, 0)` is quite different from the loss function in paper, which i think is like `[ max(0, (final1 - final2) * labelr) ]^2`. 
3. For both of the above loss function, I think gradient vanishing will occur during training, because both final1 and final2 could be very small(say 1.1e-5, 1.2e-5), and the loss will be very small. Is that the case?

Did i miss something important? 
Looking forward to your reply.  Thanks again.
